src/ethereum/utils.py

```diff
@@ -59,3 +76,354 @@ def ceil32(value: Uint) -> Uint:
         return value
     else:
         return value + ceiling - remainder
+
+
+def to_valid_address(data: bytes) -> Address:
+    """
+    Convert a byte stream to a valid address (20 bytes).
+
+    Parameters
+    ----------
+    data :
+        The string to be converted to bytes.
+
+    Returns
+    -------
+    byte_stream : `bytes`
+        The byte stream obtained from the passed string.
+    """
+    if len(data) == 20:
+        return data
+
+    if len(data) < 20:
+        return data.rjust(20, b"\x00")
+
+    return data[-20:]
```

Comment:
```suggestion
    return data[12:32]
```
seems clearer and less pythony?